### ISSUE TITLE
Getting valid headers

### ISSUE THREAD

**Issue by jamesgallagher** [dtankdempse/m3u-playlist-proxy#11]
Question for you. Firstly thanks for your work, love it all!

I am using the proxy and so far it works absolutely perfectly. I however wanted to use it for another site (nhl66.ir). I am able to get the m3u link OK but I can't work out what headers it needs. I have used developer tools and put what I think it needs, but I still just get 403s. Any tips and tricks you can suggest to get the correct headers?

**Comment by dtankdempse** [comment]
Yeah, they're using tokenized streams, which I’m sure you’re already aware of. The 403 error is likely due to a failure to rewrite the segments in the playlist from this site. I’m not entirely sure why it’s failing, so I’ll need to take a closer look. It might have something to do with the encryption key being reused multiple times throughout the playlist. I’ll check on this and get back to you. Thanks for reporting it!

**Playlist Snippet:** 
#EXTM3U
#EXT-X-VERSION:3
#EXT-X-TARGETDURATION:6
#EXT-X-PLAYLIST-TYPE:EVENT
#EXT-X-MEDIA-SEQUENCE:0
#EXT-X-PROGRAM-DATE-TIME:2024-11-25T22:29:56.350Z
#EXT-X-KEY:METHOD=AES-128,URI="http://192.168.0.87:4123?url=https%3A%2F%2Fplayback.svcs.plus.espn.com%2Fevents%2Fc902a8ed-62e9-5c3e-8445-1c77c9004a27%2Fmedia%2F873c1f82-d387-464b-9825-4052318dc17a%2Fkeys%2F14dab4f6-ba18-4493-a8a2-8cc358d3cfb3&data=QWNjZXB0PSovKnxPcmlnaW49aHR0cHM6Ly9uaGw2Ni5pcnxVc2VyLUFnZW50PU1vemlsbGEvNS4wIChXaW5kb3dzIE5UIDEwLjA7IFdpbjY0OyB4NjQ7IHJ2OjEzMi4wKSBHZWNrby8yMDEwMDEwMSBGaXJlZm94LzEzMi4w&key=true",IV=0x67C6697351FF4AEC29CDBAABF2FBE346
#EXTINF:5.005,
330/22/29/56_350.ts
#EXTINF:5.005,
330/22/30/01_355.ts
#EXTINF:5.005,
330/22/30/06_360.ts
#EXTINF:5.005,
330/22/30/11_365.ts
#EXTINF:5.005,
330/22/30/16_370.ts
#EXTINF:5.005,
330/22/30/21_375.ts
#EXTINF:5.005,
330/22/30/26_380.ts
#EXTINF:5.005,
330/22/30/31_385.ts
#EXTINF:5.005,
330/22/30/36_390.ts
#EXTINF:5.005,
330/22/30/41_395.ts
#EXTINF:5.005,
330/22/30/46_400.ts
#EXTINF:5.005,
330/22/30/51_405.ts
#EXT-X-KEY:METHOD=AES-128,URI="http://192.168.0.87:4123?url=https%3A%2F%2Fplayback.svcs.plus.espn.com%2Fevents%2Fc902a8ed-62e9-5c3e-8445-1c77c9004a27%2Fmedia%2F873c1f82-d387-464b-9825-4052318dc17a%2Fkeys%2F14dab4f6-ba18-4493-a8a2-8cc358d3cfb3&data=QWNjZXB0PSovKnxPcmlnaW49aHR0cHM6Ly9uaGw2Ni5pcnxVc2VyLUFnZW50PU1vemlsbGEvNS4wIChXaW5kb3dzIE5UIDEwLjA7IFdpbjY0OyB4NjQ7IHJ2OjEzMi4wKSBHZWNrby8yMDEwMDEwMSBGaXJlZm94LzEzMi4w&key=true",IV=0x7CC254F81BE8E78D765A2E63339FC99A
#EXTINF:5.005,
330/22/30/56_410.ts
#EXTINF:5.005,
330/22/31/01_416.ts
#EXTINF:5.005,
330/22/31/06_420.ts
#EXTINF:5.005,
330/22/31/11_426.ts
#EXTINF:5.005,
330/22/31/16_431.ts
#EXTINF:5.005,
330/22/31/21_436.ts
#EXTINF:5.005,
330/22/31/26_441.ts
#EXTINF:5.005,
330/22/31/31_446.ts
#EXTINF:5.005,
330/22/31/36_451.ts
#EXTINF:5.005,
330/22/31/41_456.ts
#EXTINF:2.268933333,
330/22/31/46_461.ts
#EXTINF:1.001,
330/22/31/48_730.ts
#EXTINF:0.767433333,
330/22/31/49_731.ts
#EXTINF:5.005,
330/22/31/50_498.ts
#EXTINF:5.005,
330/22/31/55_503.ts
